api/task_create.go

```diff
@@ -55,6 +55,15 @@ func (h *TaskLifeCycleHandler) CreateTask(w http.ResponseWriter, r *http.Request
 		return
 	}
 
+	// Initialize the new task
+	err = initNewTask(r.Context(), d)
+	if err != nil {
+		err = fmt.Errorf("error initializing new task: %s", err)
+		logger.Error("error creating task", "error", err)
+		sendError(w, r, http.StatusBadRequest, err.Error())
+		return
+	}
+
```

Comment:
I like that this got refactored out to be shared! Minor naming suggestion: What are your thoughts on the name for method `createDryRunTask()` now that it doesn't call `initNewTask()`? It seems as though it no longer creates the task and only does inspect. Maybe naming it to something like `inspectTask()`?

Reply:
That's a good point, I'll rename it!